src/librustc/middle/resolve_lifetime.rs

```diff
@@ -1464,7 +1464,17 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
             self.check_lifetime_def_for_shadowing(old_scope, &lifetime_i.lifetime);
 
             for bound in &lifetime_i.bounds {
-                self.resolve_lifetime_ref(bound);
+                if !bound.is_static() {
```

Comment:
I chose to put the common path first, even though its a negated condition. I can change if this is not preferred style though.